djangocms_bootstrap4/contrib/bootstrap4_link/cms_plugins.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+
+from cms.plugin_base import CMSPluginBase
+from cms.plugin_pool import plugin_pool
+
+from djangocms_link.cms_plugins import LinkPlugin
+from djangocms_bootstrap4.helpers import concat_classes
+
+from .models import Bootstrap4Link
+from .forms import Bootstrap4LinkForm
+
+
+class Bootstrap4LinkPlugin(LinkPlugin):
+    """
+    Components > "Button" Plugin
+    https://getbootstrap.com/docs/4.0/components/buttons/
+    """
+    model = Bootstrap4Link
+    name = _('Link / Button')
+    form = Bootstrap4LinkForm
+    change_form_template = 'djangocms_bootstrap4/admin/link.html'
+    module = _('Bootstrap 4')
+
+    LinkPlugin.fieldsets[0] = (
+        None, {
+            'fields': (
+                ('name', 'link_type'),
+                ('external_link', 'internal_link'),
+                ('link_context', 'link_size'),
+                ('link_outline', 'link_block'),
+            )
+        }
+    )
+
+    fieldsets = LinkPlugin.fieldsets
+
+    def render(self, context, instance, placeholder):
+        link_classes = []
+        if instance.link_type == 'link':
+            link_classes.append('text-{}'.format(instance.link_context))
+        else:
+            link_classes.append('btn')
+            if not instance.link_outline:
+                link_classes.append(
+                    'btn-{}'.format(instance.link_context)
+                );
+            else:
+                link_classes.append(
+                    'btn-outline-{}'.format(instance.link_context)
+                );
+        if instance.link_size:
+            link_classes.append(instance.link_size);
+        if instance.link_block:
+            link_classes.append('btn-block');
+
+        classes = concat_classes(link_classes + [
+            instance.attributes.get('class'),
+        ])
+        instance.attributes['class'] = classes
+
+        return super(Bootstrap4LinkPlugin, self).render(
+            context, instance, placeholder
+        )
+
+
+plugin_pool.unregister_plugin(LinkPlugin)
```

Comment:
I think it should be up to the developer to decide if the default link plugin should be registered or not. Maybe they both need to be in the same installation side-by-side for a while to migrate content. Or the LinkPlugin might not even be installed which would cause an exception here.

Reply:
@stefanfoulis in my tests you can still register it specifically in your application if you want so, I'd rather have it out per default when installing, that would be the 80% case.